back/app/services/side_fx_custom_field_option_service.rb

```diff
@@ -16,7 +16,10 @@ def after_update(custom_field_option, current_user)
   end
 
   def before_destroy(custom_field_option, _current_user)
-    UserCustomFieldService.new.delete_custom_field_option_values custom_field_option.key, custom_field_option.custom_field
+    custom_field = custom_field_option.custom_field
+    return unless custom_field.resource_type == 'User'
```

Comment:
The main bugfix.

Since `UserCustomFieldService.new.delete_custom_field_option_values` is only concerned with removing `User.custom_field_values` (either just the values, or the key also if the value is the only value for the key), we now return a no-op unless `custom_field.resource_type == 'User'`